integration-tests/contract-security-wrong-field-annotations.spec.ts

```diff
@@ -0,0 +1,332 @@
+      //Calling Michelson code: CONTRACT %add (pair (nat %valueA) (nat %valueB)) ;
+
+      try {
+      const op = await Tezos.contract
+        .at(contractAddition.address)
+        .then((contract) => {
+          return contract.methodsObject
+            .call({
+              entrypoint: 'Add',
+              param: {
+                valueA: 1,
+                valueB: 2,
+              },
+            })
+            .send();
+        })
+        .then((op) => {
+          return op.confirmation(3).then(() => op.hash);
+        })
+      } catch (ex) {
+        expect(ex).toEqual(expect.objectContaining({
+          message: expect.stringContaining('[a] Value is not a number: undefined')
+        }))
+      }
+    });
+
+    it('Verify call addNoAnnot', async () => {
+      const addition = await Tezos.contract.originate({
+        code: securityWrongAnnotations,
+        init: `0`,
+      });
+      await addition.confirmation();
+      const contractAddition = await addition.contract();
+
+      // Calls entrypoint: (pair %addNoAnnot nat nat))
+      // Calling Michelson code: CONTRACT %addNoAnnot (pair (nat %valueA) (nat %valueB)) ;
+      
+      try {
+      const op = await Tezos.contract
+        .at(contractAddition.address)
+        .then((contract) => {
+          return contract.methodsObject
+            .call({
+              entrypoint: 'addNoAnnot',
+              param: {
+                valueA: 1,
+                valueB: 2,
+              },
+            })
+            .send();
+        })
+        .then((op) => {
+          return op.confirmation(3).then(() => op.hash);
+        })
+      } catch (ex) {
+        expect(ex).toEqual(expect.objectContaining({
+          message: expect.stringContaining('[a] Value is not a number: undefined')
+        }))
+      }
+    });
+
+    it('Verify call AddWrongAnnot fails due to not matching field annotations', async () => {
+      const addition = await Tezos.contract.originate({
+        code: securityWrongAnnotations,
+        init: `0`,
+      });
+      await addition.confirmation();
+      const contractAddition = await addition.contract();
+
+      // Calls entrypoint: (pair %addWrongAnnot (nat %a) (nat %b))
+      // Calling Michelson code: CONTRACT %addWrongAnnot (pair (nat %valueA) (nat %valueB)) ;
+      
+      try{
+      const op = await Tezos.contract
+        .at(contractAddition.address)
+        .then((contract) => {
+          return contract.methodsObject
+            .call({
+              entrypoint: 'AddWrongAnnot',
+              param: {
+                a: 1,
+                b: 2,
+              },
+            })
+            .send();
+        })
+        .then((op) => {
+          return op.confirmation(3).then(() => op.hash);
+        })
+    } catch (ex) {
+      expect(ex).toEqual(expect.objectContaining({
+        message: expect.stringContaining('[valueA] Value is not a number: undefined')
```

Comment:
I understand the catch statement make sense in this test as the annot in the parameter are wrong

Reply:
the idea is to check what happens when the annotations I the parameter are wrong,

Reply:
Yes, for this case the annotations are wrong, so the try/catch makes sense to me. The test is correctly testing that an exception is thrown if we use bad annotation.